spec/models/user_spec.rb

```diff
@@ -0,0 +1,171 @@
+require "rails_helper"
+
+RSpec.describe User, type: :model do
+  describe "Associations" do
+    it{should have_many(:requests).dependent(:destroy)}
+    it{should have_many(:follow_books).dependent(:destroy)}
+    it{should have_many(:follow_users).dependent(:destroy)}
+    it{should have_many(:follow_authors).dependent(:destroy)}
+    it{should have_many(:comment_books_relationship).dependent(:destroy)}
+    it{should have_many(:cmt_authors_relationship).dependent(:destroy)}
+    it{should have_many(:like_books_relationship).dependent(:destroy)}
+    it{should have_many(:like_authors_relationship).dependent(:destroy)}
+    it{should have_many(:following_books)}
+    it{should have_many(:following_authors)}
+    it{should have_many(:following_users)}
+    it{should have_many(:comment_books)}
+    it{should have_many(:comment_authors)}
+    it{should have_many(:like_books)}
+    it{should have_many(:like_authors)}
+    it{should have_many(:passive_follows).dependent(:destroy)}
+    it{should have_many(:followers)}
+  end
+
+  describe "Validates" do
+    subject{FactoryBot.build(:user)}
+    context "with name field " do
+      it{should validate_presence_of(:name)}
+      it{should validate_length_of(:name)
+        .is_at_most(Settings.length.user_name_max)}
+    end
+
+    context "with email field" do
```

Comment:
phân biết describer context và it

Reply:
- describer dùng để nhóm các test case lại với nhau theo class, module, method
 - context dùng để nhóm các test case cho các điều kiện
 - it chính là các test case